### ISSUE TITLE
Label layers do not support uint64 in combination colormaps anymore with 0.4.18

### ISSUE THREAD

**Issue by thorstenwagner** [napari/napari#6077]
## 🐛 Bug

This code just runs fine in napari 0.4.17:

```python
import napari
import numpy as np
from vispy.color import Color

colors = [
    "#ff7f0e",
    "#1f77b4",
    "#2ca02c",
    "#d62728",
    "#9467bd",
    "#8c564b",
    "#e377c2",
    "#7f7f7f",
    "#bcbd22",
    "#17becf",
    "#ccebc5",
    "#ffed6f",
]
 
viewer = napari.Viewer()
img = np.arange(0,100).reshape(10,10).astype(np.uint64)

cmap = [Color(hex_name).RGBA.astype("float") / 255 for hex_name in colors]
cmap_dict = { 
                int(p): cmap[int(p) % len(cmap)]
                for p in np.unique(img)
            }
viewer.add_labels(
                        img,  
                        color=cmap_dict,  
                        name="whatever",
                    )
napari.run()
```

However, it crashes with napari 0.4.18

```
Traceback (most recent call last):
  File "/mnt/data/twagner/Projects/TomoTwin/results/202205_GeneralizationTomo/out_loss/napari_crash.py", line 28, in <module>
    viewer.add_labels(
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/components/viewer_model.py", line 4, in add_labels
    import itertools
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/labels/labels.py", line 340, in __init__
    self.color = color
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/labels/labels.py", line 556, in color
    self.color_mode = color_mode
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/labels/labels.py", line 696, in color_mode
    self.refresh()
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/base/base.py", line 1264, in refresh
    self.set_view_slice()
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/base/base.py", line 1029, in set_view_slice
    self._set_view_slice()
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/image/image.py", line 834, in _set_view_slice
    self._load_slice(data)
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/image/image.py", line 862, in _load_slice
    self._on_data_loaded(data, sync=True)
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/image/image.py", line 886, in _on_data_loaded
    if not self._slice.on_loaded(data):
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/image/_image_slice.py", line 144, in on_loaded
    self._set_raw_images(data.image, data.thumbnail_source)
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/image/_image_slice.py", line 102, in _set_raw_images
    self.image.raw = image
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/image/_image_view.py", line 78, in raw
    self._view = self.image_converter(raw_image)
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/image/image.py", line 1120, in _raw_to_displayed
    return self.obj()._raw_to_displayed(*args, **kwarg)
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/labels/labels.py", line 999, in _raw_to_displayed
    mapped_labels = self._map_labels_to_colors(labels_to_map)
  File "/opt/user_software/miniconda3_envs/napari-tomotwin/lib/python3.10/site-packages/napari/layers/labels/labels.py", line 1038, in _map_labels_to_colors
    for label_id in range(min_label_id, max_label_id + 1)
TypeError: 'numpy.float64' object cannot be interpreted as an integer
```

Was there a breaking change in 0.4.18?

Best,
Thorsten

**Comment by jni** [comment]
This was definitely not intentional @thorstenwagner, thank you for reporting. The good news is the code runs fine in #3308 which we are making a big push on, so hopefully we can get a 2-for-1 fix there (or n-for-1). Sorry about that and we hope to have an update soon! 🙏

**Comment by thorstenwagner** [comment]
Thanks for fixing that :-)

**Comment by jni** [comment]
No worries! Do watch out for #6084, although we intend to fix that before 0.5.0. 🤞
